android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java

```diff
@@ -235,86 +234,74 @@ public void getCurrentWifiSSID(final Promise promise) {
+     * Returns the RSSI (received signal strength indicator) of the currently connected WiFi network.
      */
     @ReactMethod
-    public void getCurrentSignalStrength(final Callback callback) {
-        int linkSpeed = wifi.getConnectionInfo().getRssi();
-        callback.invoke(linkSpeed);
+    public void getCurrentSignalStrength(final Promise promise) {
+        final int linkSpeed = wifi.getConnectionInfo().getRssi();
+        promise.resolve(linkSpeed);
     }
 
     /**
-     * This method will return current wifi frequency
-     *
-     * @param callback
+     * Returns the frequency of the currently connected WiFi network.
      */
     @ReactMethod
-    public void getFrequency(final Callback callback) {
-        WifiInfo info = wifi.getConnectionInfo();
-        int frequency = info.getFrequency();
-        callback.invoke(frequency);
+    public void getFrequency(final Promise promise) {
+        final WifiInfo info = wifi.getConnectionInfo();
+        final int frequency = info.getFrequency();
+        promise.resolve(frequency);
     }
 
     /**
-     * This method will return current IP
-     *
-     * @param callback
+     * Returns the IP of the currently connected WiFi network.
      */
     @ReactMethod
-    public void getIP(final Callback callback) {
-        WifiInfo info = wifi.getConnectionInfo();
-        String stringIP = longToIP(info.getIpAddress());
-        callback.invoke(stringIP);
+    public void getIP(final Promise promise) {
+        final WifiInfo info = wifi.getConnectionInfo();
+        final String stringIP = longToIP(info.getIpAddress());
+        promise.resolve(stringIP);
     }
 
     /**
-     * This method will remove the wifi network as per the passed SSID from the device list
+     * This method will remove the wifi network configuration.
+     * If you are connected to that network, it will disconnect.
      *
-     * @param ssid
-     * @param callback
+     * @param SSID wifi SSID to remove configuration for
      */
     @ReactMethod
-    public void isRemoveWifiNetwork(String ssid, final Callback callback) {
-        List<WifiConfiguration> mWifiConfigList = wifi.getConfiguredNetworks();
+    public void isRemoveWifiNetwork(final String SSID, final Promise promise) {
+        final List<WifiConfiguration> mWifiConfigList = wifi.getConfiguredNetworks();
+        final String comparableSSID = ('"' + SSID + '"'); //Add quotes because wifiConfig.SSID has them
+
         for (WifiConfiguration wifiConfig : mWifiConfigList) {
-            String comparableSSID = ('"' + ssid + '"'); //Add quotes because wifiConfig.SSID has them
             if (wifiConfig.SSID.equals(comparableSSID)) {
                 wifi.removeNetwork(wifiConfig.networkId);
                 wifi.saveConfiguration();
-                callback.invoke(true);
+                promise.resolve(true);
                 return;
             }
         }
-        callback.invoke(false);
+
+        promise.resolve(false);
```

Comment:
We should return **true** here. because the return of promise should indicate if the network has been deleted successfully. No such network has the same effect as deleting them.

Reply:
Hmmmmm. But it couldn't find it / couldn't delete it. Wouldn't it be better to reject but with an error like wifiConfigurationDoesNotExist.

Reply:
yep, it is better.